src/main/java/org/codehaus/mojo/servicedocgen/ServiceDocGenReport.java

```diff
@@ -108,6 +112,12 @@
     @Parameter( defaultValue = "false" )
     private boolean introspectFields;
 
+    @Parameter( defaultValue = "index.html, openapi.yaml" )
+    private String outputName;
```

Comment:
IMHO we should do one of the following:
* do KISS and stip this property entirely and deriving the filename from the template omitting the `.vm` extension.
* create a `Template` bean that contains the template name together with the optional `outputName` and use that instead of `String`.

If we choose the second option the config gets more complex:
```
<configuration>
  <templates>
    <template>
      <id>OpenApi.json.vm<id>
      <outputName>swagger.json</outputName>
    </template>
  </templates>
</configuration>
```

Reply:
I've implemented it as in the second option. This way we are a bit more free in choosing the file name